pkg/importer/dataStream.go

```diff
@@ -109,12 +127,12 @@ func NewDataStream(endpt, accKey, secKey string) (*dataStream, error) {
 
 // Read from top-most reader. Note: ReadFull is needed since there may be intermediate,
 // smaller multi-readers in the reader stack, and we need to be able to fill buf.
-func (d dataStream) Read(buf []byte) (int, error) {
+func (d *dataStream) Read(buf []byte) (int, error) {
```

Comment:
why pass by reference here? We are not updating the receiver? But since the byte slice has been added to the dataStream stuct it is now using more memory.

Reply:
Basically, I've always followed the pattern that if any methods have pointer receivers, they all should.  From golang faq:  https://golang.org/doc/faq#methods_on_values_or_pointers

But maybe something special is going on here that I don't get.  Like I don't understand how it would be using 512 bytes extra

If you think I should change this back or any others I changed, I will.  Just let me know.

Reply:
@mhenriks From the link above:
> First, and most important, does the method need to modify the receiver?

This is what caught my attention in the first place (plus I just fixed my bug in `matchHeader` where that method should have used a pointer to dataStream since it reads into the dataStream buf).  Continuing from the link above:
> Second is the consideration of efficiency. If the receiver is large, a big struct for instance, it will be much cheaper to use a pointer receiver.

This is why I agree that it makes sense to pass the dataStream receiver as a pointer. And, based on this reason it also makes sense the all dataStream method be by reference.